### PR TITLE
Fix some problems in the TCG searcher

### PR DESCRIPTION
**Pull request by bumbadadabum**
I feel so stupid right now.

### CODE REVIEW

File: chat-plugins/tcg.js [hunk 1]
```diff
@@ -35,12 +41,12 @@ var tcgsearch = function (target, room, user, cmd, self) {
 
 exports.commands = {
 	ygo: 'yugioh',
-	yugioh: function (target, room, user, cmd) {
+	yugioh: function (target, room, user, connection, cmd) {
```
Comment (Slayer95) [inline chat-plugins/tcg.js]: Oops!

File: chat-plugins/tcg.js [hunk 2]
```diff
@@ -20,7 +20,13 @@ var tcgsearch = function (target, room, user, cmd, self) {
 			response += data;
 		});
 		a.on('end', function () {
-			var result = JSON.parse(response);
+			var result;
+
+			try {
+				result = JSON.parse(response);
+			} catch (e) {
+				return self.sendReply("ERROR: Could not parse query:" + e);
```
Comment (Slayer95) [inline chat-plugins/tcg.js]: I'm not sure if it's a good idea to send the error details as a reply to the user.
Maybe log to the console?

Comment (bumbadadabum) [inline chat-plugins/tcg.js]: If I put in a command and it doesn't work, I would want to know it doesn't work and what the error is. Idk I can change it easily.

Comment (Zarel) [inline chat-plugins/tcg.js]: This is fine; `response` isn't exactly privileged/secret data or anything.

**Pull request merged**